packages/main/src/TimePickerClock.ts

```diff
@@ -0,0 +1,1000 @@
+import UI5Element from "@ui5/webcomponents-base/dist/UI5Element.js";
+import customElement from "@ui5/webcomponents-base/dist/decorators/customElement.js";
+import property from "@ui5/webcomponents-base/dist/decorators/property.js";
+// import slot from "@ui5/webcomponents-base/dist/decorators/slot.js";
+import event from "@ui5/webcomponents-base/dist/decorators/event.js";
+// import type { ClassMap } from "@ui5/webcomponents-base/dist/types.js";
+import litRender from "@ui5/webcomponents-base/dist/renderer/LitRenderer.js";
+
+import Integer from "@ui5/webcomponents-base/dist/types/Integer.js";
+
+// Template
+import TimePickerClockTemplate from "./generated/templates/TimePickerClockTemplate.lit.js";
+
+// Styles
+import TimePickerClockCss from "./generated/themes/TimePickerClock.css.js";
+
+type TimePickerClockItem = {
+	angle?: number,
+	item?: string,
+	innerItem?: string,
+}
+
+type TimePickerClockSelection = {
+	showMarker: boolean,
+	itemClasses?: string,
+	innerItemClasses?: string,
+}
+
+type TimePickerClockDimensions = {
+	radius: number,
+	centerX: number,
+	centerY: number,
+	dotHeight: number,
+	numberHeight: number,
+	outerMax: number,
+	outerMin: number,
+	innerMax: number,
+	innerMin: number,
+	offsetX: number,
+	offsetY: number,
+}
+
+type TimePickerClockSelectedItem = TimePickerClockItem & TimePickerClockSelection;
+
+const ANIMATION_DURATION_MAX = 200; // total animation duration, without the delay before firing the event
+const ANIMATION_DELAY_EVENT = 100; // delay before firing the event
+const LONG_TOUCH_DURATION = 1000; // duration for long-touch interaction
+const CLOCK_ANGLE_STEP = 6;
+const CLOCK_NUMBER_CLASS = "ui5-tp-clock-number";
+const CLOCK_NUMBER_HOVER_CLASS = "ui5-tp-clock-number-hover";
+const CLOCK_NUMBER_SELECTED_CLASS = "ui5-tp-clock-selected";
+const CLOCK_MIDDOT_CLASS = "ui5-tp-clock-mid-dot";
+
+/**
+ * @class
+ *
+ * <h3 class="comment-api-title">Overview</h3>
+ *
+ * <code>ui5-time-picker-clock</code> is the single clock item to use inside a <code>ui5-time-picker-clocks</code>.
+ * <code>ui5-time-picker-clocks</code> is one of the internal controls of <code>ui5-time-picker</code>
+ *
+ * <h3>Usage</h3>y
+ *
+ * <code>ui5-time-picker-clock</code> can display hours, minutes or seconds
+ *
+ * <h3>ES6 Module Import</h3>
+ *
+ * <code>import "@ui5/webcomponents/dist/TimePickerClock.js";</code>
+ *
+ * @constructor
+ * @author SAP SE
+ * @alias sap.ui.webc.main.TimePickerClock
+ * @extends sap.ui.webc.base.UI5Element
+ * @abstract
+ * @tagname ui5-time-picker-clock
+ * @since 1.??.??
+ * @public
```

Comment:
New public component ui5-time-picker-clock, is this intended?
I thought we redesign the `ui5-time-picker`

Reply:
Err, sorry, my mistake, it will be private...
In general, it is part of another, also private control, that will replace sliders.